django_opensearch_dsl/documents.py

```diff
@@ -115,14 +105,41 @@ def get_indexing_queryset(
             if verbose:
                 stdout.write(f"{action} {model}: {round(i / count * 100)}% ({self._eta(start, done, count)})\r")
 
-            for obj in qs[i : i + chunk_size]:
+            for obj in qs.iterator(chunk_size=chunk_size):
                 done += 1
+                if done % chunk_size == 0:
+                    stdout.write(f"{action} {model}: {round(done / count * 100)}% ({self._eta(start, done, count)})\r")
                 yield obj
 
-            i = min(i + chunk_size, count)
-
-        if verbose:
-            stdout.write(f"{action} {count} {model}: OK          \n")
+    def get_indexing_queryset(
+        self,
+        db_alias: str = None,
+        verbose: bool = False,
+        filter_: Optional[Q] = None,
+        exclude: Optional[Q] = None,
+        count: int = None,
+        action: OpensearchAction = OpensearchAction.INDEX,
+        stdout: io.FileIO = sys.stdout,
+        batch_size: int = None,
+    ) -> Iterable:
+        """Divide the queryset into chunks."""
+        qs = self.get_queryset(db_alias=db_alias, filter_=filter_, exclude=exclude, count=count)
+        if self._order_indexing_queryset and not qs.query.is_ordered:
+            qs = qs.order_by("pk")
+        if batch_size:
+            # In order to avoid loading big querysets into memory or
+            # loading them in temporary tables in the database,
+            # we have the possibility to divide the queryset using batch_size.
+            result = qs.aggregate(min_pk=Min("pk"), max_pk=Max("pk"))
+            min_value = result["min_pk"]
+            max_value = result["max_pk"] + 1
+            for pk_offset in range(min_value, max_value, batch_size):
+                max_pk = min(pk_offset + batch_size, max_value)
+                batch_qs = qs.filter(pk__gte=pk_offset, pk__lt=max_pk)
+                stdout.write(f"Processing batch with pk from {pk_offset} to {max_pk - 1}\n")
+                yield from self._yield_objects(batch_qs, action, verbose, stdout)
```

Comment:
We lose the global ETA here. You should move the ETA from `_yield_objects` back to `get_indexing_queryset` (this means converting the `yield from ...` back to a `for` loop to allow enumeration).